### PR TITLE
Factor out module loading to a separate place

### PR DESCRIPTION
**Pull request by dbkr**
So it can be reused

### CODE REVIEW

File: synapse/config/password_auth_providers.py [hunk 1]
```diff
@@ -15,13 +15,15 @@
...
 
-import importlib
+from synapse.util.module_loader import load_module
 
 
 class PasswordAuthProviderConfig(Config):
     def read_config(self, config):
         self.password_providers = []
 
+        provider_config = None
```
Comment (erikjohnston) [inline synapse/config/password_auth_providers.py]: Why are we doing this?

File: synapse/util/module_loader.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    provider_class = getattr(module, clz)
+
+    try:
+        provider_config = provider_class.parse_config(provider["config"])
+    except Exception as e:
+        raise ConfigError(
+            "Failed to parse config for %r: %r" % (provider['module'], e)
+        )
+
+    return (provider_class, provider_config)
```
Comment (erikjohnston) [inline synapse/util/module_loader.py]: Technically no parens are required

File: synapse/config/password_auth_providers.py (hunk 1 above)
Comment (dbkr) [reply synapse/config/password_auth_providers.py]: just to make it clear the variable is at this level  (even though they all get hoisted)

Comment (erikjohnston) [reply synapse/config/password_auth_providers.py]: we don't usually bother, but ok

**Pull request merged**